### ISSUE TITLE
TS >=4.1: Cannot assign because it is not a variable (require does not work with let anymore)

### ISSUE THREAD

**Issue by sk-** [microsoft/TypeScript#41628]
Starting with TS 4.1 and also in TS 4.2, it seems that required code is frozen and cannot be reassigned.

We had some code that would require a JSON and conditionally overwrite it, however that code is no longer valid. See example below for a minimal reproduction.

I couldn't find anything in the release notes or [breaking changes](https://devblogs.microsoft.com/typescript/announcing-typescript-4-1/#breaking-changes).

**TypeScript Version:**  4.1.2

**Search Terms:**
require json cannot assign let

**Code**

```js
let a = require('./package.json');
a = {...a};
```

**Expected behavior:**
No warnings should be raised for this code. Alternatively at the least I would expect another warning about defining `a` with `let`.

**Actual behavior:**
```
 Cannot to 'a' because it is not a variable.
```

**Playground Link:** 
https://www.typescriptlang.org/play?ts=4.2.0-dev.20201121&useJavaScript=true#code/DYUwLgBAhhC8ECcQEcCuBLJAKA5AOgHoAHKAYwGsoBzEPAKwGcB7AOxwEoBuAWACgZ4AbzwioAXz5A

**Related Issues:**

**Comment by DanielRosenwasser** [comment]
This might need to be ported back to a 4.1.3 release.

**Comment by a-tarasyuk** [comment]
I think that's because TS binds `let x = require('y')` statements the same way as `import x = require('y');` - as an alias - https://github.com/microsoft/TypeScript/pull/39770. Not sure, maybe the binder should only set the alias flag for const variables
https://github.com/microsoft/TypeScript/blob/d070acf29d579bf0afbfd97a4941bf9171966305/src/compiler/binder.ts#L3305-L3306

**Comment by sandersn** [comment]
I looked at the frequency of `const`, `let` and `var` in conjunction with `require` and found that `const` outnumbers `let` about 100 to 1. `var` outnumbers let more than 200 to 1. Unfortunately, I couldn't think of an easy way to find re-assignments without compiling all the code I searched through. So I looked at the user tests, which *are* compiled. But it didn't have any errors like this.

I can think of two possible fixes:

1. Correctly allow assignments to both `let` and `var`, since both are technically mutable (although it's rare in practice).
2. Allow assignment to `let` only, since it's rare and it lets us keep the improved declaration emit and goto-def for `var`, which is more common.

And of course there's always:
0. Do nothing.

(1) causes us to lose aliasing for `var`, which is a big loss. (2) makes another corner of JS handling more complex. Since this only affects projects with `checkJs: true` and use `var`/`let` with `require` to mutate the binding, I think the best course is to do nothing. I would recommend `// @ts-ignore` for cases where you need to write this in a checked JS file.

Side note: I don't think we've noted breaking changes to checkJs features in the past. We need to improve that. And of course I just missed that this would be a break at the time.
